### PR TITLE
use binarybuilder/provider cross compilation

### PR DESCRIPTION
**Pull request by SimonDanisch**
Uses BinaryBuilder with this script to host and cross compile binaries: https://github.com/SimonDanisch/ClipperBuilder

https://github.com/JuliaPackaging/BinaryProvider.jl
https://github.com/JuliaPackaging/BinaryBuilder.jl/

### CODE REVIEW

**Comment by SimonDanisch** [comment]
Cross compiled binaries are hosted here versioned by tags:
https://github.com/SimonDanisch/ClipperBuilder/tree/master/products

This should make installation much easier, since no compilation tool chain needs to be present

File: src/Clipper.jl [hunk 1]
```diff
@@ -19,12 +19,11 @@ module Clipper
...
-        const library_path = joinpath(dirname(@__FILE__), "cclipper.dll")
-    end
-
-    @static if is_unix()
-        const library_path = joinpath(dirname(@__FILE__), "cclipper.so")
+    if isfile("../deps/deps.jl")
+        include("../deps/deps.jl")
+        check_deps()
+    else
+        error("Clipper.jl not correctly build. Please run Pkg.build(\"Clipper\") and try again!")
```
Comment (maxvoxel8) [inline src/Clipper.jl]: should be `built` not `build`

**Comment by maxvoxel8** [comment]
I guess it makes me a bit nervous because it means whoever is maintaining Clipper.jl can't easily update the clipper version without updating the hosted libraries which are managed by somebody else. Is there a reason why the built libraries can't be hosted in the same repo?

Also, I'm not really working with Julia anymore. If you'd like, I can transfer the repo to you and you can make whatever changes you want.

**Comment by SimonDanisch** [comment]
>I guess it makes me a bit nervous because it means whoever is maintaining Clipper.jl can't easily update the clipper version without updating the hosted libraries which are managed by somebody else

Sure that's sub optimal. Ideally, they should be in the same org! If you want, we can just put both packages in JuliaGeometry.

A different repository is used for the following reasons:

* BinaryProvider actually supports automatic builds with travis, which I haven't activated yet because of some bugs - but then it needs to be in a different repo

* release cycles for binaries and Clipper.jl might be different - so you don't want to host new binaries for every tag.

**Comment by maxvoxel8** [comment]
Ok, I will transfer to JuliaGeometry.

**Comment by maxvoxel8** [comment]
It's giving me an error that JuliaGeometry/Clipper.jl already exists. Perhaps you could just change METADATA.jl to point at the JuliaGeometry repo and that would make yours the "official" one?

**Comment by SimonDanisch** [comment]
Ok I will try that!

**Comment by SimonDanisch** [comment]
Now that https://github.com/JuliaLang/METADATA.jl/pull/15558 is through, should we just delete my fork and transfer this repository to JuliaGeometry?

**Comment by SimonDanisch** [comment]
So i think this PR would need to get merged first, and then we should be ready to do this!

**Comment by maxvoxel8** [comment]
I think once I merge the PR, you will need to delete your repo so I can transfer this one to JuliaGeometry.

**Comment by SimonDanisch** [comment]
yes, that's the plan!

**Comment by SimonDanisch** [comment]
Do you have the propper rights to move the package to JuliaGeometry already?

**Comment by maxvoxel8** [comment]
The error I get when I try to transfer is "JuliaGeometry/Clipper.jl already exists and You don’t have the permission to create repositories on JuliaGeometry." So maybe not.

**Comment by SimonDanisch** [comment]
Ok, let me give you the rights and delete the repository right now!

**Comment by SimonDanisch** [comment]
Sent out the invite! So I'll go ahead and delete it ;)

**Comment by SimonDanisch** [comment]
Actually I just transferred it to my account, just in case ;) Hope it works!

**Comment by maxvoxel8** [comment]
Now I get the error, "You can only transfer a repository from an organization to yourself at this time". So I think you will have to create a new Clipper.jl repo and push from local to initialize it.

**Comment by SimonDanisch** [comment]
Or transfer it to yourself and then JuliaGeometry?

**Comment by maxvoxel8** [comment]
Ok, that seems to have worked.

**Pull request merged**